include/modem/ntn.h

```diff
@@ -0,0 +1,130 @@
+ *       - mfw_nrf9151-ntn
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief NTN event types.
+ */
+enum ntn_evt_type {
+	/**
+	 * Modem location update request.
+	 *
+	 * This event indicates whether the modem needs location updates. Modem needs to know
+	 * the current location when NTN is used. When location updates are requested, the
+	 * ntn_modem_location_set() function must be called to set the current location to the
+	 * modem. The event contains the accuracy requested by the modem. When the event is
+	 * received with @c nrf_modem_location.updates_requested set to true, the
+	 * ntn_modem_location_set() function should be called as soon as possible even
+	 * if location has been already set to the modem, because the modem may have requested
+	 * location with better accuracy.
+	 *
+	 * The required interval for calling the function depends on the requested accuracy and
+	 * speed of the device. The accuracy of the location should remain within the requested
+	 * accuracy at all times. If the device can move at a maximum of 120 km/h (33.3 m/s),
+	 * the location should be updated at least every six seconds. It's the responsibility of
+	 * the application to maintain the location with good enough accuracy at all times. If
+	 * the modem does not have the current location with sufficient accuracy, NTN usage is
+	 * not possible.
+	 *
+	 * The associated payload is the @c ntn_evt.modem_location member of type
+	 * @ref ntn_modem_location in the event.
+	 */
+	NTN_EVT_MODEM_LOCATION_UPDATE = 0,
+};
+
+/**
+ * @brief Location update request.
+ */
+struct ntn_modem_location {
+	/** If true, location updates are requested by the modem. */
+	bool updates_requested;
+
+	/** Accuracy (in meters) requested by the modem. */
+	uint32_t accuracy;
+};
+
+/**
+ * @brief NTN event.
+ */
+struct ntn_evt {
+	/** Event type. */
+	enum ntn_evt_type type;
+
+	/** Event data. */
+	union {
+		/** Payload for event @ref NTN_EVT_MODEM_LOCATION_UPDATE. */
+		struct ntn_modem_location modem_location;
+	};
+};
+
+/**
+ * @brief Event handler prototype.
+ *
+ * @param[in] evt Pointer to event structure.
+ */
+typedef void (*ntn_evt_handler_t)(const struct ntn_evt *evt);
+
+/**
+ * @brief Register an event handler.
+ *
+ * @param[in] handler Event handler. Can be NULL to unregister the handler.
+ */
+void ntn_register_handler(ntn_evt_handler_t handler);
+
+/**
+ * @brief Set location to the modem.
+ *
+ * This function is used to set the location to the modem for NTN usage.
+ *
+ * @param[in] latitude Latitude in degrees.
+ * @param[in] longitude Longitude in degrees.
+ * @param[in] altitude Altitude in meters.
+ * @param[in] validity Location validity in seconds. 0 means infinite validity.
+ *
+ * @retval 0 Location was set successfully.
+ * @retval -EFAULT AT command could not be sent to the modem.
+ */
+int ntn_modem_location_set(double latitude, double longitude, float altitude, uint32_t validity);
```

Comment:
You must have thought about whether to put these parameters as a structure. I'm just thinking that we are lacking one parameter from the AT command and one day in few months, we will need that accuracy also into the parameter of this function...